### PR TITLE
Issue 1697 crypto 8.7.0

### PR DESCRIPTION
**Pull request by IanBullard**
This adds Crypto++ 8.7.0 to xrepo

### CODE REVIEW

File: packages/l/libcurl/versions.lua [hunk 1]
```diff
@@ -53,6 +53,4 @@ function add_versions_list()
     add_versions("7.33.0", "0afde4cd949e2658eddc3cda675b19b165eea1af48ac5f3e1ec160792255d1b3")
     add_versions("7.32.0", "8e3db42548e01407cb2f1407660c0f528b89ec7afda6264442fc2b229b95223b")
     add_versions("7.31.0", "a73b118eececff5de25111f35d1d0aafe1e71afdbb83082a8e44d847267e3e08")
-
-    add_patches("7.84.0", path.join(os.scriptdir(), "patches", "7.84.0", "sched.patch"), "e79f56f840cbc6996a153f19d9266bd46fe4154e6b494c8ee0478cb5b87662d3")
```
Comment (IanBullard) [inline packages/l/libcurl/versions.lua:57]: I'll pull this out, somehow it got into this branch.

File: packages/c/cryptopp/xmake.lua [hunk 2]
```diff
@@ -17,7 +19,20 @@ package("cryptopp")
     on_install("windows", "macosx", "linux", "bsd", "iphoneos", function (package)
         local cryptopp_cmake = package:resourcedir("cryptopp_cmake")
         os.cp(path.join(cryptopp_cmake, "*", "CMakeLists.txt"), ".")
-        os.cp(path.join(cryptopp_cmake, "*", "cryptopp-config.cmake"), ".")
+        local version = package:version()
+        
+        if (version:ge("8.7")) then
+            if (package:config("shared")) then
+                print("ERROR: CryptoC++ 8.7+ does not support shared libraries")
```
Comment (IanBullard) [inline packages/c/cryptopp/xmake.lua:26]: @waruqi I don't know if this is appropriate, let me know if I should handle this differently

File: packages/c/cryptopp/xmake.lua [hunk 3]
```diff
@@ -32,6 +47,13 @@ package("cryptopp")
     end)
 
     on_test(function (package)
+        local version = package:version()
+        if (version:ge("8.7")) then
+            if (package:config("shared")) then
+                -- cryptopp_cmake for 8.7 does not support shared libraries so there's nothing to test
+                return
```
Comment (IanBullard) [inline packages/c/cryptopp/xmake.lua:54]: @waruqi Same here.

File: packages/c/cryptopp/xmake.lua [hunk 4]
```diff
@@ -17,7 +19,20 @@ package("cryptopp")
     on_install("windows", "macosx", "linux", "bsd", "iphoneos", function (package)
         local cryptopp_cmake = package:resourcedir("cryptopp_cmake")
         os.cp(path.join(cryptopp_cmake, "*", "CMakeLists.txt"), ".")
-        os.cp(path.join(cryptopp_cmake, "*", "cryptopp-config.cmake"), ".")
+        local version = package:version()
+        
+        if (version:ge("8.7")) then
+            if (package:config("shared")) then
+                print("ERROR: Crypto++ 8.7+ does not support shared libraries")
+                return
```
Comment (waruqi) [inline packages/c/cryptopp/xmake.lua:27]: you can disable shared, see

https://github.com/xmake-io/xmake-repo/blob/766154c7806aca96c87500f2e84b70c9dd1e1b5b/packages/m/mma/xmake.lua#L11

Comment (IanBullard) [reply packages/c/cryptopp/xmake.lua:27]: Thank you, I'll make this fix tomorrow.

**Pull request closed without merging**